### PR TITLE
add enabled to product

### PR DESCRIPTION
**Pull request by brunoparma88**
EM-323(https://rootstrap.atlassian.net/browse/EM-323)
Create enabled flag to projects

### CODE REVIEW

File: app/admin/products.rb [hunk 1]
```diff
@@ -1,13 +1,14 @@
 ActiveAdmin.register Product do
   permit_params :id, :description, :name, :jira_key, :new_jira_key,
-                jira_board_attributes: %i[id jira_project_key],
-                repository_ids: []
+                :enabled, jira_board_attributes: %i[id jira_project_key],
+                          repository_ids: []
```
Comment (sebastiancaraballo) [inline app/admin/products.rb:4]: Is alignment okay?

File: db/migrate/20220412181602_add_enabled_to_products.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class AddEnabledToProjects < ActiveRecord::Migration[6.0]
+  def change
+    change_table :products do |t|
+      t.boolean :enabled, null: false, default: true
```
Comment (sebastiancaraballo) [inline db/migrate/20220412181602_add_enabled_to_products.rb:4]: Maybe it could be worth adding an index

File: app/models/product.rb [hunk 3]
```diff
@@ -24,7 +25,7 @@ class Product < ApplicationRecord
   has_many :metrics, as: :ownable, dependent: :destroy
 
   validates :name, presence: true, uniqueness: true
-
+  validates :enabled, presence: true
```
Comment (sebastiancaraballo) [inline app/models/product.rb:28]: Would it be a good idea to set `enabled` at the default_scope? Just asking. https://apidock.com/rails/ActiveRecord/Base/default_scope/class

File: app/admin/products.rb (hunk 1 above)
Comment (brunoparma88) [reply app/admin/products.rb:4]: Rubocop put it there.

File: app/models/product.rb (hunk 3 above)
Comment (brunoparma88) [reply app/models/product.rb:28]: I didn't know about default_scope, it's a good idea.

File: app/models/product.rb [hunk 4]
```diff
@@ -5,26 +5,28 @@
...
 #
 #  index_products_on_deleted_at  (deleted_at)
+#  index_products_on_enabled     (enabled)
 #  index_products_on_name        (name)
 #
 
 class Product < ApplicationRecord
   acts_as_paranoid
-
+  scope :enabled, lambda {
```
Comment (sebastiancaraballo) [inline app/models/product.rb:22]: This should go at the end. Take a look at [this](https://github.com/rootstrap/tech-guides/blob/master/ruby/rails.md#macro-style-methods)

**Review by sebastiancaraballo — APPROVED** [review]

File: app/models/product.rb [hunk 5]
```diff
@@ -5,26 +5,28 @@
 #  id          :bigint           not null, primary key
 #  deleted_at  :datetime
 #  description :string
+#  enabled     :boolean          default(TRUE), not null
```
Comment (lauperalti) [inline app/models/product.rb:8]: Could we add presence validation of the enabled attribute in the model and in spec?

File: app/models/alert.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@
 # Table name: alerts
 #
 #  id             :bigint           not null, primary key
-#  active         :boolean
+#  active         :boolean          default(FALSE), not null
```
Comment (lauperalti) [inline app/models/alert.rb:6]: Could we add presence validation of the `active` attribute in the model and in spec?

**Review by lauperalti — APPROVED** [review]
LGTM. Left some minor comments. 🚀

File: app/models/alert.rb (hunk 6 above)
Comment (brunoparma88) [reply app/models/alert.rb:6]: the field "activate" doesn't change in this PR. Maybe annotate was not run in the previous PR

**Pull request merged**